lib/ansible/modules/windows/win_get_url.ps1

```diff
@@ -3,10 +3,14 @@
 # Copyright: (c) 2015, Paul Durivage <paul.durivage@rackspace.com>
 # Copyright: (c) 2015, Tal Auslander <tal@cloudshare.com>
 # Copyright: (c) 2017, Dag Wieers <dag@wieers.com>
+# Copyright: (c) 2019, Viktor Utkin <viktor_utkin@epam.com>
+# Copyright: (c) 2019, Uladzimir Klybik <uladzimir_klybik@epam.com>
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
 #AnsibleRequires -CSharpUtil Ansible.Basic
 #Requires -Module Ansible.ModuleUtils.AddType
+#Requires -Module Ansible.ModuleUtils.FileUtil
+#Requires -Module Ansible.ModuleUtils.Legacy
```

Comment:
I wouldn't re-add `Legacy` now we are using the `Ansible.Basic` format. Just take a snippet of the hash function and add it here. It does lead to code duplication but at least we aren't copying a whole bunch of code that's not needed.

Reply:
Hi @jborean93 , we need to use function Get-FileChecksum for checksum calculation from Ansible.ModuleUtils.Legacy.
Any suggestion about any other way of getting checksum?

Reply:
Direct usage of the .Net class [HashAlgorithm](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.hashalgorithm?redirectedfrom=MSDN&view=netframework-4.5) like described in the post [Use PowerShell To Calculate the Hash of a File](https://learn-powershell.net/2013/03/25/use-powershell-to-calculate-the-hash-of-a-file/)?

Reply:
What I meant was to just copy a subset of that function in Legacy.psm1 into this module. This is similar to a recent change to win_stat where we moved away from Legacy.psm1 and had to do this https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/windows/win_stat.ps1#L16-L34.

It's not ideal but Legacy is legacy and we want to move away from modules using it eventually. In the future we could add the hashing mechanism to another module_util (or it's own) but I'm still looking into how that will pan out.

Reply:
Hi @jborean93 , Do you think that implement a hash function in every module is a good practice? So if in any day you will want to add some functionality to it, you should support all modules where we implement it. You don't like legacy module, I am ok with this, but for this you should move  `Get-FileChecksum` in another shared lib/include file.

Reply:
@it-praktyk This function should be shared between some win modules, I don't want to implement it in every module.

Reply:
There is a `Get-FileHash` function in powershell, but it works only with version great than 3.

Reply:
Yes it should be shared but it is only shared in Legacy.psm1 which is wrong. I agree code duplication is bad but for now I'm happy taking that hit to move off Legacy.psm1.

In the future the file hashing work could be in it's own util or included with FileUtil but I'm still playing around that space so I don't want to hold you up on it.

Reply:
Legacy removed.